src/webui/www/private/scripts/client.js

```diff
@@ -230,62 +230,40 @@ window.addEventListener("DOMContentLoaded", () => {
     buildLogTab();
     MochaUI.initializeTabs("mainWindowTabsList");
 
+    setStatusFilter = function(name) {
+        selectedStatus = name;
+        highlightSelectedStatus();
+        updateMainData();
+        LocalPreferences.set("selected_filter", name);
```

Comment:
I'm having second second thoughts about this. Perhaps this should be moved to the top within this function. The reason is that `highlightSelectedStatus()` and `updateMainData()` could be slow and therefore we should save the preference first. I was assuming the functions was so slow that the user got impatient and closed the webui and we didn't saved the last preference.